frontend/src/components/modals/accountDelete/useDeleteAccountModal.tsx

```diff
@@ -1,11 +1,32 @@
+import { useMutation } from "@tanstack/react-query";
+import { IRequestError } from "helpers/error-interface";
+import { deleteUser } from "services/UsersService";
+import useToken from "hooks/useToken";
+import { FormEvent } from "react";
+import useFetchUser from "hooks/useFetchUser";
+
 export const useDeleteAccountModal = (hideModal: () => void) => {
-    const deleteAccountConfirm = () => {
-        //TODO: account delete logic
-        hideModal();
+    const { id } = useFetchUser();
+    const { setAccessToken } = useToken();
+
+    const mutation = useMutation<null, IRequestError>({
+        mutationFn: () => deleteUser(id!),
+        onSuccess: () => {
+            setAccessToken("");
+            hideModal();
+        }
+    });
+    const deleteAccountConfirm = (event: FormEvent) => {
+        event.preventDefault();
+
+        if (id) {
```

Comment:
More space is saved if u use shortify 'if' statement
```suggestion
        id && ....
```